File: CloudMusic/src/main/java/com/ken/android/CloudMusic/Fragment/BaseFragment.java
package com.ken.android.CloudMusic.Fragment;

import android.app.Fragment;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import org.xutils.x;

/**
 * Created by axnshy on 16/8/9.
 */
public class BaseFragment extends Fragment {
    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container, Bundle savedInstanceState) {
        x.view().inject(this,inflater,container);
        return super.onCreateView(inflater, container, savedInstanceState);
    }
}

File: SmartPlay/src/main/java/com/luffy/smartplay/ui/base/BaseFragment.kt
package com.luffy.smartplay.ui.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.fragment.app.Fragment
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider
import androidx.viewbinding.ViewBinding

abstract class BaseFragment<T : ViewBinding,M: ViewModel> : Fragment() {
    abstract val viewModel: M
    lateinit var viewBinding: T

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return bindViewBinding(inflater, container).apply {
            viewBinding = this
        }.root
    }

    abstract fun bindViewBinding(inflater: LayoutInflater,
                                      container: ViewGroup?) : T
}